components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp

```diff
@@ -309,7 +310,7 @@
                                                     <% if (!action.equals("")) { %>
                                                     <a class="icon-link"
                                                        style="background-image:url(../admin/images/edit.gif);"
-                                                       href='addOperationParameter.jsp?editparam=editparam&operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=action%>&queryId=<%=selectedQueryId%>&paramNameId=<%=aWithParamsList.getName()%>&operationParamId=<%=(aWithParamsList.getParamValue())%>'>
+                                                       href='addOperationParameter.jsp?editparam=editparam&operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=Encode.forHtml(action)%>&queryId=<%=selectedQueryId%>&paramNameId=<%=aWithParamsList.getName()%>&operationParamId=<%=(aWithParamsList.getParamValue())%>'>
```

Comment:
forUriComponent should be used
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String-